js/src/evaluation/evaluate_comparative.ts

```diff
@@ -125,10 +141,34 @@ export async function evaluateComparative(
   }
 
   const client = options.client ?? new Client();
+  const resolvedExperiments = await Promise.all(experiments);
 
-  const projects = await Promise.all(
-    experiments.map((experiment) => loadExperiment(client, experiment))
-  );
+  const projects = await (() => {
+    if (!isExperimentResultsList(resolvedExperiments)) {
+      return Promise.all(
+        resolvedExperiments.map((experiment) =>
+          loadExperiment(client, experiment)
+        )
+      );
+    }
+
+    // if we know the number of runs beforehand, check if the
+    // number of runs in the project matches the expected number of runs
+    return Promise.all(
+      resolvedExperiments.map((experiment) =>
+        pRetry(
+          async () => {
+            const project = await loadExperiment(client, experiment);
+            if (project.run_count !== experiment?.results.length) {
```

Comment:
What if the experiment was deliberately only run over a subset of the runs?

Reply:
That should be fine, as we are only comparing the local experiment state with the server state